### PR TITLE
Fix local up cluster dns with RBAC

### PR DESCRIPTION
**Pull request by deads2k**
The DNS server needs permissions to run using RBAC.  This does it with a big hammer before we sort out tight permissions.

@fabianofranz for the CLI change
@xilabao for the local-up-cluster change

### CODE REVIEW

**Comment by fabianofranz** [comment]
CLI changes LGTM.

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -742,18 +749,15 @@ function start_kubedns {
...
+        
+        # TODO update to dns role once we have one.
+        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" create clusterrolebinding system:kube-dns --clusterrole=cluster-admin --serviceaccount=kube-system:default
         # use kubectl to create skydns rc and service
-        ${KUBECTL} --namespace=kube-system create -f skydns-rc.yaml
-        ${KUBECTL} --namespace=kube-system create -f skydns-svc.yaml
+        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f skydns-rc.yaml
+        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f skydns-svc.yaml
         echo "Kube-dns rc and service successfully deployed."
+        rm  skydns-rc.yaml skydns-svc.yaml namespace.yaml
```
Comment (xilabao) [inline hack/local-up-cluster.sh]: remove namespace.yaml, use `rm  skydns-rc.yaml skydns-svc.yaml`
the others LGTM

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -401,6 +401,14 @@ contexts:
     name: local-up-cluster
 current-context: local-up-cluster
 EOF
+
+    # flatten the kubeconfig files to make them self contained
```
Comment (sttts) [inline hack/local-up-cluster.sh]: no need for the temporary file. Just pipe the HERE document above through `kubectl config view --kubeconfig /dev/stdin --minify --flatten`

Comment (deads2k) [inline hack/local-up-cluster.sh]: > no need for the temporary file. Just pipe the HERE document above through kubectl config view --kubeconfig /dev/stdin --minify --flatten

I tried here: https://gist.github.com/deads2k/18d7d352671fbb50d2c362a326eae489, but the kubeconfigs were always empty.

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (deads2k) [inline hack/local-up-cluster.sh]: > emove namespace.yaml, use rm skydns-rc.yaml skydns-svc.yaml
> the others LGTM

fixed

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (sttts) [inline hack/local-up-cluster.sh]: leave it as it is for now. Will take a look later.

**Pull request merged**